nix/solidity-package.sh

```diff
@@ -1,20 +1,14 @@
 source $stdenv/setup
 unpackPhase
 
-jsonopts=--combined-json=abi,bin,bin-runtime,srcmap,srcmap-runtime,ast
+jsonopts=--combined-json=abi,bin,bin-runtime,srcmap,srcmap-runtime,ast,metadata
 
 export DAPP_SRC=$src
 export DAPP_OUT=out
 
-find "$DAPP_SRC" -name '*.sol' | while read -r x; do
-  dir=${x%\/*}
-  dir=${dir#$DAPP_SRC}
-  dir=${dir#/}
-  mkdir -p "$DAPP_OUT/$dir"
-  (set -x; solc --overwrite $REMAPPINGS --allow-paths $DAPP_SRC $solcFlags --abi --bin --bin-runtime -o "$DAPP_OUT/$dir" "$x")
-  json_file=$DAPP_OUT/$dir/${x##*/}.json
-  (set -x; solc $REMAPPINGS --allow-paths $DAPP_SRC $solcFlags $jsonopts "$x" >"$json_file")
-done
+mkdir -p "$DAPP_OUT"
+mapfile -t files < <(find "${DAPP_SRC}" -name '*.sol')
+(set -x; solc "$REMAPPINGS" "$jsonopts" "$solcFlags" /=/ "${files[@]}" > "${DAPP_JSON}")
```

Comment:
Where is `DAPP_JSON` declared?

Reply:
Right, I'm hardcoding it to `dapp.sol.json` now. If that is not an option as generates conflicts for different packages we will define something different.